strawberry/codegen/query_codegen.py

```diff
@@ -519,7 +522,14 @@ def _convert_variable_definitions(
                 variable_type,
             )
 
-            type_.fields.append(GraphQLField(variable.name, None, variable_type))
+            type_.fields.append(
+                GraphQLField(
+                    variable.name,
+                    None,
+                    variable_type,
+                    definition_node=variable_definition,
```

Comment:
should this have `strawberry_field` as well?

Reply:
Hmmm ... originally, I didn't think so ... but now that I think about it a little more, possibly ...

This represents the input variables to the operation.  I guess this is represented by the query/mutation operation object?

Let me see what I can cook up.

Reply:
just for more context, I'd like to keep as abstracted as possible 😊 we might change the internal representation (ie moving away from GraphQL) in future

Reply:
So ... the more I dig into this, the more that this feels weird (or at least really complex).  For example, you could have a query like:

```graphql
query ($skipTitle: Boolean!, $val: Int!) {
  queryPost(val: $val) {
    id
    title @skip(if: $skipTitle)
    text
  }
}
```

In this case, the `$skipTitle` variable field really has nothing to do with any of the types in the backend schema definition.  The `val` field presumably _does_, but it's buried in the `Query.query_post` field's base_resolver (I guess you'd introspect the wrapped_func on that resolver to understand the backend metadata about this thing?)

For now, I think that I'll make the `graphql_type` on the `{type}Variables` point at the `Query` or `Mutation` class and then users can crawl that class however they want in their plugins.  If we ever see more concrete use-cases, we can make better decisions on how we want to set these pieces of metadata to support those use-cases at that point.

Does that sound reasonable to you?